week_2/pet.rb

```diff
@@ -110,6 +116,13 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    pets_by_type = pets.group_by { |pet| pet.animal_type_id }
+    by_animal_type = {}
+    pets_by_type.each do |animal_type_id, pets|
+      pet_names = []
+      pets.each { |pet| pet_names << pet.name }
```

Comment:
`map` can use used here
```suggestion
      pet_names = pets.map { |pet| pet.name } 
```

Reply:
Hey, I have made the changes.